fe/fe-core/src/main/java/com/starrocks/qe/ShowExecutor.java

```diff
@@ -666,13 +667,28 @@ private void handleShowCreateTable() throws AnalysisException {
             }
 
             if (table instanceof View) {
+                if (showStmt.getTableType() == Table.TableType.MATERIALIZED_VIEW) {
+                    ErrorReport.reportAnalysisException(ErrorCode.ERR_WRONG_OBJECT, showStmt.getDb(),
+                            showStmt.getTable(), "MATERIALIZED VIEW");
+                }
                 View view = (View) table;
                 StringBuilder sb = new StringBuilder();
                 sb.append("CREATE VIEW `").append(table.getName()).append("` AS ").append(view.getInlineViewDef());
                 rows.add(Lists.newArrayList(table.getName(), createTableStmt.get(0), "utf8", "utf8_general_ci"));
                 resultSet = new ShowResultSet(ShowCreateTableStmt.getViewMetaData(), rows);
+            } else if (table instanceof MaterializedView) {
```

Comment:
could you explain this logic?

Reply:
sales_records->tableType:Table
sales_records_view->tableType:VIEW
store_amt_mv->tableType:MATERIALIZED_VIEW
case:
show create table sales_records;->ok
show create view sales_records;->report error sales_records is not a view
show create materialized view sales_records;->report error sales_records is not a materialized view

show create table sales_records_view;->ok
show create view sales_records_view;->ok
show create materialized view sales_records_view;->report error sales_records_view is not a materialized view

show create table store_amt_mv;->ok
show create view store_amt_mv;->report error store_amt_mv is not a view
show create materialized view store_amt_mv;->ok